python/ray/_private/gcs_aio_client.py

```diff
@@ -0,0 +1,143 @@
+import logging
+from typing import Dict, List, Optional
+from concurrent.futures import ThreadPoolExecutor
+from ray._raylet import GcsClient
+from ray.core.generated import (
+    gcs_service_pb2,
+)
+import ray._private.utils
+
+
+logger = logging.getLogger(__name__)
+
+
+class AsyncProxy:
+    def __init__(self, inner, loop, executor):
+        self.inner = inner
+        self.loop = loop
+        self.executor = executor
+
+    def _function_to_async(self, func):
+        async def wrapper(*args, **kwargs):
+            return await self.loop.run_in_executor(self.executor, func, *args, **kwargs)
+
+        return wrapper
+
+    def __getattr__(self, name):
+        """
+        If attr is callable, wrap it into an async function.
+        """
+        attr = getattr(self.inner, name)
+        if callable(attr):
+            return self._function_to_async(attr)
+        else:
+            return attr
+
+
+class GcsAioClient:
+    def __init__(
+        self,
+        loop=None,
+        executor=None,
+        address: Optional[str] = None,
+        nums_reconnect_retry: int = 5,
+    ):
+        if loop is None:
+            loop = ray._private.utils.get_or_create_event_loop()
+        if executor is None:
+            # TODO(ryw): make this number configurable
+            executor = ThreadPoolExecutor(
+                max_workers=1, thread_name_prefix="gcs_aio_client"
+            )
+
+        self._gcs_client = GcsClient(address, nums_reconnect_retry)
+        self._async_proxy = AsyncProxy(self._gcs_client, loop, executor)
+        self._connect()
+        self._nums_reconnect_retry = nums_reconnect_retry
+
+    def _connect(self):
+        self._gcs_client._connect()
+
+    @property
+    def address(self):
+        return self._gcs_client.address
+
+    async def check_alive(
+        self, node_ips: List[bytes], timeout: Optional[float] = None
+    ) -> List[bool]:
+        logger.debug(f"check_alive {node_ips!r}")
+        return await self._async_proxy.check_alive(node_ips, timeout)
```

Comment:
can you add a simple test if this is actually async?

Reply:
(something like you add a rpc with asyncio.sleep(2) and use asyncio.gather to call multiple of them, make sure the whole gather doesn't take no more than 2 seconds?)